lib/ext.js

```diff
@@ -145,21 +132,27 @@ module.exports = class Ext {
         }
 
         // Removes pagination from query parameters if default is true
-        // If defaults is false, paginaton param is needed in the generated links
+        // If defaults is false, pagination param is needed in the generated links
         if (this.config.query.pagination.default) {
             delete request.query[this.config.query.pagination.name];
         }
 
         const source = request.response.source;
-        const results = Array.isArray(source) ? source : source[this.config.reply.parameters.results.name];
+        const results = Array.isArray(source)
+            ? source
+            : source[this.config.reply.parameters.results.name];
         Hoek.assert(Array.isArray(results), 'The results must be an array');
 
-        const baseUri = this.config.uri + request.url.pathname + '?';
+        const baseUri = `${this.config.uri + request.url.pathname}?`;
         const query = request.query; // Query parameters
         const currentPage = query[this.config.query.page.name];
         const currentLimit = query[this.config.query.limit.name];
+        const { zeroIndex } = this.config;
+        const pageIndex = zeroIndex ? 0 : 1;
```

Comment:
I would call this `startIndex`

Reply:
Changed this to "firstPage" to A) align with the "lastPage" change below, and B) because "startIndex" is in use elsewhere.